finesse/hardware/plugins/time/ntp_time.py

```diff
@@ -49,23 +54,33 @@ def __init__(
         self._ntp_port = ntp_port
         self._ntp_timeout = ntp_timeout
 
-    def close(self) -> None:
-        """Close the connection to the device."""
-
-    def get_time(self) -> float:
-        """Get the current time.
+        # Set up time offset polling.
+        self.poll_time_offset()
+        self._poll_timer = QTimer()
+        self._poll_timer.timeout.connect(self.poll_time_offset)
+        self._poll_timer.start(int(ntp_poll_interval * 1000))
 
-        Returns:
-            The current time in seconds since the epoch.
-        """
+    def poll_time_offset(self) -> None:
+        """Query the NTP server for the current time offset."""
         try:
-            response = self._client.request(
+            self._response = self._client.request(
                 self._ntp_host,
                 version=self._ntp_version,
                 port=self._ntp_port,
```

Comment:
I've only just noticed that this is a string, which seems a bit weird. I'm guessing the string `"ntp"` is just an alias for the default port (123). If so, I'd just use the number. Also, the typing does matter here, because the frontend casts device parameters to whatever type it thinks they should be and I'm guessing that a string value of e.g. `"1234"` won't work.